src/android/usb_host/android_uvc.cpp

```diff
@@ -0,0 +1,1804 @@
+            frame->intervals[i] = DW_TO_INT(p);
+            p += 4;
+        }
+        frame->intervals[block[25]] = 0;
+    }
+
+    DL_APPEND(format->frame_descs, frame);
+
+    return UVC_SUCCESS;
+}
+
+/** @internal
+* Process a single VideoStreaming descriptor block
+* @ingroup device
+*/
+uvc_error_t usbhost_uvc_parse_vs(
+        usbhost_uvc_device *dev,
+        usbhost_uvc_device_info_t *info,
+        usbhost_uvc_streaming_interface_t *stream_if,
+        const unsigned char *block, size_t block_size) {
+    uvc_error_t ret;
+    int descriptor_subtype;
+
+    ret = UVC_SUCCESS;
+    descriptor_subtype = block[2];
+
+    switch (descriptor_subtype) {
+        case UVC_VS_INPUT_HEADER:
+            ret = usbhost_uvc_parse_vs_input_header(stream_if, block, block_size);
+            break;
+        case UVC_VS_FORMAT_UNCOMPRESSED:
+            ret = usbhost_uvc_parse_vs_format_uncompressed(stream_if, block, block_size);
+            break;
+        case UVC_VS_FORMAT_MJPEG:
+            ret = usbhost_uvc_parse_vs_format_mjpeg(stream_if, block, block_size);
+            break;
+        case UVC_VS_FRAME_UNCOMPRESSED:
+        case UVC_VS_FRAME_MJPEG:
+            ret = usbhost_uvc_parse_vs_frame_uncompressed(stream_if, block, block_size);
+            break;
+        case UVC_VS_FORMAT_FRAME_BASED:
+            ret = usbhost_uvc_parse_vs_frame_format(stream_if, block, block_size);
+            break;
+        case UVC_VS_FRAME_FRAME_BASED:
+            ret = usbhost_uvc_parse_vs_frame_frame(stream_if, block, block_size);
+            break;
+        case UVC_VS_COLORFORMAT:
+            break;
+
+        default:
+            /** @todo handle JPEG and maybe still frames or even DV... */
+            /*fprintf(stderr, "unsupported descriptor subtype: %d\n",
+            descriptor_subtype);*/
+            break;
+    }
+
+    return ret;
+}
+
+uvc_error_t update_stream_if_handle(usbhost_uvc_device *devh, int interface_idx) {
+    usbhost_uvc_streaming_interface_t *stream_if;
+
+    DL_FOREACH(devh->deviceData.stream_ifs, stream_if) {
+        if (stream_if->bInterfaceNumber == interface_idx)
+            if (!stream_if->interfaceHandle.device && interface_idx < MAX_USB_INTERFACES) {
+                // usbhost_GetAssociatedInterface returns the associated interface (Video stream interface which is associated to Video control interface)
+                // A value of 0 indicates the first associated interface (Video Stream 1), a value of 1 indicates the second associated interface (video stream 2)
+                // WinUsbInterfaceNumber is the actual interface number taken from the USB config descriptor
+                // A value of 0 indicates the first interface (Video Control Interface), A value of 1 indicates the first associated interface (Video Stream 1)
+                // For this reason, when calling to usbhost_GetAssociatedInterface, we must decrease 1 to receive the associated interface
```

Comment:
I don't understand how this comment is relevant to what's going on in the code here?

Reply:
Beats me! It looks like copy-pasted stuff from WinUSB code is here too?